applications/SaaS3.md

```diff
@@ -0,0 +1,318 @@
+
+  - The Creator receives the royalty fee as the reward for their commitment to the docker development; they are the author of Web2 API codes. 
+
+![image](https://user-images.githubusercontent.com/95557343/159788719-f6fb785d-65ce-4f7c-a3cb-5b0226708895.png)
+
+
+##### Security & Off-chain Miner Consensus 
+
+- Miner security - To qualify, a miner or node is required to stake a particular amount of tokens as collateral (POS). Each request for API will trigger an on-chain zero-knowledge proof check (zkc), which determines whether the node is rewarded or punished. 
+
+- API security - Similar to API3 insurance, API security is achieved via Delegation Proof of Staking (DPoS). Particularly, the API creator is required to stake an amount of tokens at the time of submitting the dAPI deployment. The amount is set at a default value in their first submission and is adjusted in proportion to the service income. Besides, the token holders are able to stake on the dAPIs as a collateral and share profits with API creator. The mechanism is established following the same vein of the concepts in Polkadot / Kusama, including staking, nomination, and validation.
+ 
+The process described above establishes a de facto safeguard mechanism, sheltering dAPI users from potential malfunctions of the computational node or the API service, and when available providing remedies in compensation. 
+
+##### Collateral and Sue Claims
+In the case of a malfunction, the user detects the misperformance and assesses the damages and losses, and is able to lodge a sue claim published on the Court DAO. In supplement, the staked tokens or collaterals are 'locked' for a period, a mechanism to discourage stakers from 'front-running claims', i.e. withdrawing the staked token as soon as a miner or service malfunctions to evade claims. On the other hand, the litigant is required to stake funds to be able to make a claim, which elevates the cost of abuse. The Court DAO deliberates on the sue claim and decides whether a remedy should be made to the victimized dAPI user. Once a claim is passed in Court DAO, the amount of tokens as has been claimed will be transferred to the litigant in compensation. 
+
+The same mechanism also applies to the punishment meted out for the node(s) that malfunctioned with on-chain zkc. Such a reward-and-punishment mechanism encourages nodes to act in good faith and better service quality. Hence, a much lower risk of potential cyber threats such as the Sybil attack.
+
+##### Implementation Layers
+
+![image](https://user-images.githubusercontent.com/95557343/155807244-b10e6228-edf9-479a-add9-b4c347f62e2e.png)
+
+#### Technologies
+
+1. Substrate
+2. Docker
+3. Virtualization Technology
+4. Deep learning frameworks
+5. libp2p
+6. IPFS / ARWeave
+7. gRPC
+8. Rust
+9. Substrate front-end template
+10. Go
```

Comment:
Just out of curiosity, which component(s) are you going to develop in Go?

Reply:
@takahser Great thanks for your approval and suggestions. But since the pull request was closed, no chance to apply your advice. If there is modifications of the project in the future, we will pull an update request to commit your suggestions and future changes together. Actually, Docker is written in the Go language. If it is necessary, we will try to add an extension beyond docker source code, but hopefully not. In addition, although this is the level2 application, could you please introduce some VCs for us. We are seeking for a seed round. Thanks again for your review efforts.

Reply:
@SaaS3DAO sounds great, thank you for elaborating on this! Looking forward to your delivery!